### ISSUE TITLE
feat: tool use

### ISSUE THREAD

**Issue by aarnphm** [yetone/avante.nvim#372]
Currently, we are doing one-shot prompting.

Since most of the providers we support (Claude, openai, cohere, Gemini) all support tool calls, it is probably good to add an option to integrate these.

one thing I want to do is:
- websearch

**Comment by ianpaul10** [comment]
Hey! I'm a huge fan of avante and looking to contribute. I see you assigned yourself but I was wondering if you were still planning to work on it or if you'd be open to me taking a look at it.

**Comment by aarnphm** [comment]
Yes this is still in the works atm. I'm currently a bit busy with life so I'm slowing working towards this

**Comment by ianpaul10** [comment]
Ok, does that mean you want help working on it? Or should I look at other issues

**Comment by aarnphm** [comment]
oh let me finish this one, feel free to take other issues

**Comment by dogonthehorizon** [comment]
Another use case I see enabled is being able to create tools that RAG over code/documentation dependencies rather than just the current workspace enabled by `@codebase`. Motivating example is teaching the LLM how to use internal libraries that aren't part of the training set without having to copy/paste definitions/context between repositories
